pkg/image/cdi/validate.go

```diff
@@ -3,21 +3,56 @@ package cdi
 import (
 	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
 	"github.com/harvester/harvester/pkg/image/backend"
+	"github.com/harvester/harvester/pkg/image/common"
 	"github.com/harvester/harvester/pkg/webhook/types"
 )
 
 type Validator struct {
+	vmiv common.VMIValidator
 }
 
-func GetValidator() backend.Validator {
-	return &Validator{}
+func GetValidator(vmiv common.VMIValidator) backend.Validator {
+	return &Validator{vmiv}
 }
 
-func (cv *Validator) Create(_ *types.Request, _ *harvesterv1.VirtualMachineImage) error {
+func (cv *Validator) Create(req *types.Request, vmImg *harvesterv1.VirtualMachineImage) error {
+	if err := cv.vmiv.CheckDisplayName(vmImg); err != nil {
+		return err
+	}
+
+	if err := cv.vmiv.CheckURL(vmImg); err != nil {
```

Comment:
What do we think of adding [`url.Parse(vmi.Spec.URL)`](https://pkg.go.dev/net/url#Parse) in `CheckURL()` to make sure it's well-formed URL?

Reply:
ditto, updated the enhancement